server/master/src/main/java/org/apache/accumulo/master/tableOps/bulkVer2/PrepBulkImport.java

```diff
@@ -179,37 +181,23 @@ private void checkForMerge(final Master master) throws Exception {
     VolumeManager fs = master.getFileSystem();
     final UniqueNameAllocator namer = master.getContext().getUniqueNameAllocator();
     Path sourceDir = new Path(bulkInfo.sourceDir);
-    FileStatus[] files = fs.listStatus(sourceDir);
+    List<FileStatus> files = BulkImport.filterInvalid(fs.listStatus(sourceDir));
 
     Path bulkDir = createNewBulkDir(master.getContext(), fs, bulkInfo.tableId);
     Path mappingFile = new Path(sourceDir, Constants.BULK_LOAD_MAPPING);
 
     Map<String,String> oldToNewNameMap = new HashMap<>();
-    for (FileStatus file : files) {
-      final FileStatus fileStatus = file;
-      final Path originalPath = fileStatus.getPath();
-      String[] fileNameParts = originalPath.getName().split("\\.");
-      String extension = "";
-      boolean invalidFileName;
-      if (fileNameParts.length > 1) {
-        extension = fileNameParts[fileNameParts.length - 1];
-        invalidFileName = !FileOperations.getValidExtensions().contains(extension);
-      } else {
-        invalidFileName = true;
-      }
-      if (invalidFileName) {
-        log.warn("{} does not have a valid extension, ignoring", fileStatus.getPath());
-        continue;
-      }
 
-      String newName = "I" + namer.getNextName() + "." + extension;
-      Path newPath = new Path(bulkDir, newName);
-      oldToNewNameMap.put(originalPath.getName(), newPath.getName());
+    for (FileStatus file : files) {
+      // since these are only valid files we know it has an extension
+      String newName =
+          "I" + namer.getNextName() + "." + FilenameUtils.getExtension(file.getPath().getName());
+      oldToNewNameMap.put(file.getPath().getName(), new Path(bulkDir, newName).getName());
     }
 
     // also have to move mapping file
-    Path newMappingFile = new Path(bulkDir, mappingFile.getName());
-    oldToNewNameMap.put(mappingFile.getName(), newMappingFile.getName());
+    oldToNewNameMap.put(mappingFile.getName(), new Path(bulkDir, mappingFile.getName()).getName());
```

Comment:
seems like this could be further simplified like :

```java
oldToNewNameMap.put(mappingFile.getName(), mappingFile.getName());
```

Reply:
except the latter is a "new" directory path "bulkdir"

Reply:
Yeah I think @keith-turner is correct.  The map should just contain the name of the mappingFile.  Since it won't change, it will just be the same name again.  It looks strange but it should work with the next step, which is the move.

Reply:
But I don't think the bulkDir matters.  I wrote the following test program

```java
public class Test {
  public static void main(String[] args) {
    Path mappingFile = new Path("/some/source", Constants.BULK_LOAD_MAPPING);
    
    System.out.println("name 1 : " + mappingFile.getName());
    
    new Path("/some/bulk", mappingFile.getName()).getName();
    
    System.out.println("name 2 : " +  new Path("/some/bulk", mappingFile.getName()).getName());
  }
}
```

Which prints :

```
name 1 : loadmap.json
name 2 : loadmap.json
```